hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/RecordListBuilder.java

```diff
@@ -133,9 +133,10 @@ private SingleTransactionRecordBuilder doAddChild(@NonNull final Configuration c
             throw new IndexOutOfBoundsException("No more child slots available");
         }
 
-        final var consensusNow =
-                recordBuilders.get(childCount - 1).consensusNow().plusNanos(1L);
+        final var previousRecord = recordBuilders.get(childCount - 1);
+        final var consensusNow = previousRecord.consensusNow().plusNanos(1L);
         final var recordBuilder = new SingleTransactionRecordBuilder(consensusNow);
+        recordBuilder.parentConsensusTimestamp(previousRecord.parentConsensusTimestamp());
```

Comment:
If we set the `parentConsensusTimestamp` of the top level record (i.e. the first one) to null (see my other comment) here for the first child we should use `previousRecord.consensusNow()` and for all the other children we can then use the `previousRecord.parentConsensusTimestamp()`.

Reply:
That is (unfortunately) correct. We could simplify this again in the future, but for now, we have to differentiate these two cases.